### ISSUE TITLE
Build trending hashtags in to left nav

### ISSUE THREAD

**Issue by RichKavanagh** [diaspora/diaspora#6012]
I (and no doubt many others) have a feature request.

I'd  love to see trending hashtags be listed in the left nav.

There are various ways to do it and there is no right or wrong way, it merely depends on ones definition of the word 'trending'. What it should *not* be, is simply a list of 'popular' tags.

If you're building this in to core I'd recommend 3x podmin level options in diaspora.yml to allow the podmin to turn on or off: Local Trending Tags, Worldwide Trending Tags, Popular Tags

(Where 'popular' **is** a list of popular (by count))

Then allow the users to choose in their profile settings the same 3x options.

The reason to allow or differentiate between local and worldwide is due to the size of the pod. A pod with a small amount users will  struggle to find trending tags.

Both myself and Brian õ have put in countless hours on this elsewhere so can happily provide core logic / flow process examples where required.

Can't code it myself as I know zero ruby :/

**Comment by RichKavanagh** [comment]
Just to add, another 'nice to have' is directional arrows showing if the tag is trending upwards or downwards. Plus you must trim (...) long tags after *x* chrs to stop them overflowing in to the main stream / content area.

**Comment by jhass** [comment]
https://github.com/biola/punching_bag might allow for a nice and flexible implementation that can potentially be decoupled from pure "x usages in y time"

**Comment by RichKavanagh** [comment]
Do you guys have a specific format / document for specs or can I just post a flow chart type diagram? :)

**Comment by jhass** [comment]
What the tags means is that there are many ways to approach and implement such a feature and that we should decide on some concrete mechanics and technologies before approaching it ;)

**Comment by RichKavanagh** [comment]
Yup, agreed. Again that comes back to each persons definition of the word 'trending'. So should that converstation take place here or over at Loomio?

**Comment by jhass** [comment]
Loomio might indeed be better.

**Comment by RichKavanagh** [comment]
I'll get something typed up in draft over the weekend :+1:

**Comment by RichKavanagh** [comment]
Discussion started at https://www.loomio.org/d/9YpxctHk/build-trending-hashtags-in-to-left-nav

I'll reply back here with a spec in a few weeks :)
